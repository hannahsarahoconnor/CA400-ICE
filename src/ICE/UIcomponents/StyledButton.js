 
import React from 'react';
import { StyleSheet, Text, Alert, View, Image, TouchableOpacity, ImageBackground, Button } from 'react-native';

const styledButton = props => {
    const content = (
        <View style={[styles.button, {backgroundColor: props.color}]}>
        <Text style={styles.text}>{props.text}</Text>
        </View>
    )

   return <TouchableOpacity onPress={props.onPress}>{content}</TouchableOpacity>
}

const styles = StyleSheet.create({
    button: {
        borderRadius:20,
        borderWidth:2,
        padding: 5,
        width: 300,
        borderColor: "#4277a7"
      },
    text: {
        color: "white",
        fontSize: 35,
        alignSelf: "center",
        fontWeight: 'bold'
    }
});

export default styledButton;

