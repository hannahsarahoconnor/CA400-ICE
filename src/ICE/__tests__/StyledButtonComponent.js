import React from 'react';
import { shallow } from 'enzyme';
import StyledButton from "../UIcomponents/StyledButton";

describe('StyledButton', () => {
  describe('Rendering', () => {
      it('should match to snapshot - Primary', () => {
          const component = shallow(<StyledButton label="test label" primary />)
          expect(component).toMatchSnapshot("Primary button snapshot")
      });
      it('should match to snapshot - Secondary', () => {
          const component = shallow(<StyledButton label="test label" primary={false} />)
          expect(component).toMatchSnapshot("Secondary button snapshot")
      });
  });
});