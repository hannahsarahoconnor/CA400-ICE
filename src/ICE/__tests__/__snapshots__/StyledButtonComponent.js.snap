// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton Rendering should match to snapshot - Primary: Primary button snapshot 1`] = `ShallowWrapper {}`;

exports[`StyledButton Rendering should match to snapshot - Secondary: Secondary button snapshot 1`] = `ShallowWrapper {}`;
